File: TPII29_09/Gof/Estruturais/Adapter/Adapter01.java
// Interface do Cliente
interface ITarget {
    void request();
}

// Cliente
class Cliente {
    private ITarget target;

    public Cliente(ITarget target) {
        this.target = target;
    }

    public void makeRequest() {
        System.out.println("Fazendo Uma Requisição");
        target.request();
    }
}

// Serviço Existente (Adaptee)
class Adaptee {
    public void specificRequest() {
        System.out.println("Requisição específica do Adaptee");
    }
}

// Adaptador (Adapter)
class Adapter implements ITarget {
    private Adaptee adaptee;

    public Adapter(Adaptee adaptee) {
        this.adaptee = adaptee;
    }

    public void request() {
        adaptee.specificRequest();
    }
}

// Classe Main para execução
public class Adapter01 {
    public static void main(String[] args) {
        Adaptee adaptee = new Adaptee();
        Adapter adapter = new Adapter(adaptee);
        Cliente cliente = new Cliente(adapter);

        cliente.makeRequest();
    }
}

File: TPII29_09/Gof/Estruturais/Adapter/Adapter01.py
# Interface do Cliente
class ITarget:
    def request(self):
        pass

# Cliente
class Cliente:
    def __init__(self, target):
        self.target = target

    def make_request(self):
        print("Fazendo Uma Requisição")
        self.target.request()

# Serviço Existente (Adaptee)
class Adaptee:
    def specific_request(self):
        print("Requisição específica do Adaptee")

# Adaptador (Adapter)
class Adapter(ITarget):
    def __init__(self, adaptee):
        self.adaptee = adaptee

    def request(self):
        self.adaptee.specific_request()

# Utilizando o Adaptador
adaptee = Adaptee()
adapter = Adapter(adaptee)
cliente = Cliente(adapter)

cliente.make_request()
